adapters/coap/src/test/java/org/eclipse/hono/adapter/coap/CoapContextTest.java

```diff
@@ -162,4 +169,69 @@ void testStartAckTimerHandlesNonNumberPropertyValue() {
         ctx.startAcceptTimer(vertx, tenant, 500);
         verify(vertx).setTimer(eq(500L), VertxMockSupport.anyHandler());
     }
+
+    /**
+     * Verifies that the CoAP time option is not in the response if not requested by either a request option or parameter.
+     */
+    @Test
+    void testTimeOptionIsNotIncludedInResponseIfNotRequested() {
+        final CoapExchange exchange = mock(CoapExchange.class);
+        final Adapter coapConfig = new Adapter(Constants.PROTOCOL_ADAPTER_TYPE_COAP);
+        final TenantObject tenant = TenantObject.from("tenant", true).addAdapter(coapConfig);
+        final var authenticatedDevice = new DeviceUser(tenant.getTenantId(), "device-id");
+        final CoapContext ctx = CoapContext.fromRequest(exchange, authenticatedDevice, authenticatedDevice, "4711", span);
+        final Response response = mock(Response.class);
+        final OptionSet responseOptions = new OptionSet();
+        when(response.getOptions()).thenReturn(responseOptions);
+        ctx.respond(response);
```

Comment:
you seem to implicitly rely on the fact that the mocked CoapExchange's *getRequestOptions()* method returns `null`. FMPOV we should configure the mock explicitly instead, e.g. returning an empty set of Options, as you did with the response.